src/internal/arg.c

```diff
@@ -80,49 +80,43 @@ r_obj* ffi_enquo(r_obj* sym, r_obj* frame) {
 
 // Match ------------------------------------------------------------------
 
-r_obj* ffi_arg_match0(r_obj* args) {
-  args = r_node_cdr(args);
-
-  r_obj* arg = r_node_car(args); args = r_node_cdr(args);
-  r_obj* values = r_node_car(args); args = r_node_cdr(args);
-  r_obj* env = r_node_car(args);
-
+// [[ export() ]]
+int arg_match(r_obj* arg, r_obj* values, r_obj* arg_nm) {
   if (r_typeof(arg) != R_TYPE_character) {
-    arg_match0_abort("`%s` must be a character vector.", env);
+    r_abort("`%s` must be a character vector.", unwrap_c_str(arg_nm));
   }
   if (r_typeof(values) != R_TYPE_character) {
     r_abort("`values` must be a character vector.");
   }
 
-  r_ssize arg_len = r_length(arg);
-  r_ssize values_len = r_length(values);
+  int arg_len = r_length(arg);
+  int values_len = r_length(values);
```

Comment:
I return an `int` because it's meant to be coerced to `enum` by the caller. We could still use `r_ssize` internally as you suggest, but it feels a bit unnecessary given that this function is not going to be called with long arrays.